### ISSUE TITLE
Meteor 1.6 and bcrypt package

### ISSUE THREAD

**Issue by conradj** [fractal-code/meteor-azure#46]
This is a modified Todo app, which builds fine on my Mac but I can't deploy to Azure.

- Operating system 
MacOS 10.12.6

- Meteor version
```cat .meteor/release```
METEOR@1.6
```meteor show meteor```
Package: meteor@1.8.0 - I just updated to this one from 1.7.1. because meteor-azure failed when trying to install NPM so I've upgraded to latest.

- Package version (run `meteor-azure -v`)
v2.0.7

- Command output with debug mode enabled
```
debug:   completed Azure authentication
info:    Updating Azure application settings
debug:   set environment variables
debug:   set Meteor settings
debug:   set Kudu deployment settings
debug:   set Node to v8.8.1
debug:   set NPM to v5.4.2
info:    Compiling application bundle
debug:   generate meteor build
warn:    Using default web config
debug:   create tarball
info:    Deploying bundle tarball
info:    Running server initialisation, polling for status...
debug:   Retrieving Kudu deployment log...
debug:   Command: bash script.sh
debug:   meteor-azure: Now using NVM v1.1.4
debug:   meteor-azure: Setting Node version
debug:   Version 8.8.1 is already installed.
debug:   meteor-azure: Now using Node v8.8.1 (32-bit)
debug:   meteor-azure: Now using NPM v5.4.2
debug:   meteor-azure: Unpacking bundle
debug:   meteor-azure: Using default web config
debug:   meteor-azure: Setting Node runtime
debug:   meteor-azure: Enabling IISNode logging
debug:   meteor-azure: Installing NPM dependencies
debug:   /tmp/8d520269dffcf4f/bundle/programs/server /tmp/8d520269dffcf4f
debug:
debug:   > fibers@2.0.0 install D:\local\Temp\8d520269dffcf4f\bundle\programs\server\node_modules\fibers
debug:   > node build.js || nodejs build.js
debug:
debug:   `win32-ia32-57` exists; testing
debug:   Binary is fine; exiting
debug:
debug:   > meteor-dev-bundle@0.0.0 install D:\local\Temp\8d520269dffcf4f\bundle\programs\server
debug:   > node npm-rebuild.js
debug:
debug:   npm WARN saveError EPERM: operation not permitted, rename 'D:\local\Temp\8d520269dffcf4f\bundle\programs\server\npm-shrinkwrap.json.1605601119' -> 'D:\local\Temp\8d520269dffcf4f\bundle\programs\server\npm-shrinkwrap.json'
debug:   added 124 packages in 26.644s
debug:   npm WARN meteor-dev-bundle@0.0.0 No description
debug:   /tmp/8d520269dffcf4f
debug:   npm WARN meteor-dev-bundle@0.0.0 No repository field.
debug:   npm WARN meteor-dev-bundle@0.0.0 No license field.
debug:   meteor-azure: Rebuilding NPM dependencies
debug:   /tmp/8d520269dffcf4f/bundle/programs/server/npm /tmp/8d520269dffcf4f
debug:
debug:
debug:   > bcrypt@1.0.3 install D:\local\Temp\8d520269dffcf4f\bundle\programs\server\npm\node_modules\bcrypt
debug:   > node-pre-gyp install --fallback-to-build
debug:
debug:   'node-pre-gyp' is not recognized as an internal or external command,
debug:   operable program or batch file.
debug:   npm ERR! code ELIFECYCLE
debug:   npm ERR! errno 1
debug:   npm ERR! bcrypt@1.0.3 install: `node-pre-gyp install --fallback-to-build`
debug:   npm ERR! Exit status 1
debug:   npm ERR!
debug:   npm ERR! Failed at the bcrypt@1.0.3 install script.
debug:   npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
debug:
debug:   npm ERR! A complete log of this run can be found in:
debug:   npm ERR!     D:\local\AppData\npm-cache\_logs\2017-10-31T06_16_41_559Z-debug.log
debug:   meteor-azure: Could not rebuild NPM dependencies
debug:   npm WARN saveError EPERM: operation not permitted, rename 'D:\local\Temp\8d520269dffcf4f\bundle\programs\server\npm-shrinkwrap.json.1605601119' -> 'D:\local\Temp\8d520269dffcf4f\bundle\programs\server\npm-shrinkwrap.json'\r\nnpm WARN meteor-dev-bundle@0.0.0 No description\r\nnpm WARN meteor-dev-bundle@0.0.0 No repository field.\r\nnpm WARN meteor-dev-bundle@0.0.0 No license field.\r\n\r\n'node-pre-gyp' is not recognized as an internal or external command,\r\noperable program or batch file.\r\nnpm ERR! code ELIFECYCLE\r\nnpm ERR! errno 1\r\nnpm ERR! bcrypt@1.0.3 install: `node-pre-gyp install --fallback-to-build`\r\nnpm ERR! Exit status 1\r\nnpm ERR! \r\nnpm ERR! Failed at the bcrypt@1.0.3 install script.\r\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\r\n\r\nnpm ERR! A complete log of this run can be found in:\r\nnpm ERR!     D:\local\AppData\npm-cache\_logs\2017-10-31T06_16_41_559Z-debug.log\r\nmeteor-azure: Could not rebuild NPM dependencies\r\nD:\Program Files (x86)\SiteExtensions\Kudu\67.61027.3099\bin\Scripts\starter.cmd bash script.sh
error:   Failed to complete server initialisation
```
- Log from Kudu
[2017-10-31T06_16_41_559Z-debug.log](https://github.com/fractal-code/meteor-azure/files/1429736/2017-10-31T06_16_41_559Z-debug.log)

Thank you :-)

**Comment by ramijarrar** [comment]
I suspect this may be package-specific (though we haven't tested with Meteor 1.6 yet) - if this isn't a sensitive codebase, are you comfortable sharing access for reproduction?

**Comment by conradj** [comment]
[Sure, here it is](https://github.com/conradj/todos/tree/totterup)

**Comment by ramijarrar** [comment]
I was able to reproduce the error on one of our test servers and the issue is isolated to the bcrypt package, which doesn't appear to compile properly for Windows on Meteor 1.6. I'm pretty swamped with work at the moment, so won't get a chance to take a closer look for the next few weeks.

In the meantime though, you should be able to deploy successfully by temporarily removing the bcrypt package (see my fork [here](https://github.com/fractal-code/todos/tree/totterup)) or downgrading back to Meteor 1.5. Let me know if you have any issues with either approach.

**Comment by conradj** [comment]
Thanks - have removed bcrypt for now and it deployed fine. If I deployed with your modified web.config the app doesn't work - just thought I'd mention it in case it was important!

**Comment by ramijarrar** [comment]
@conradj 

Haven't had time to start working on this yet, but just for reference there seems to be some similar reports for 1.6 users in-general https://github.com/meteor/meteor/issues/9162 https://github.com/peerlibrary/node-fiber-utils/issues/3

**Comment by ramijarrar** [comment]
This has been fixed as of v2.0.8 (along with some other compatibility improvements for Meteor 1.6)

Please let me know if you have any further issues adding bcrypt/other packages and functionality
